### PR TITLE
Fixes #172 : Add handling of mongoengine.errors.validation error

### LINKED ISSUE #172: [back] add handling of mongoengine.errors.ValidationError when there is not a valid object id in the request
too much info in the title

### PR DESCRIPTION
**Pull request by Draeggon**
Fix #172

### CODE REVIEW

**Comment by Draeggon** [comment]
for now only the rooms are done i wait for your return before doing the other res

**Review by Benvii — CHANGES_REQUESTED** [review]
Try to use a MethodView decorator instead of calling it in each resource should be easier to integrate on all resources.

File: server/scratchy_server/resources/roomres.py [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+from scratchy_server.resources import func
 
 
 @doc(tags=['Room'])
 @marshal_with(RoomSchema)
 class RoomRes(MethodResource):
 
     def get(self, roomId):
-        return RoomModel.objects().get_or_404(id=roomId)
+        return func.get(RoomModel, roomId)
```
[the remark this answers is not in the document]
Comment (Draeggon) [reply server/scratchy_server/resources/roomres.py:12]: i don't really understand the doc could you make a more understandable example ?

File: server/scratchy_server/filters/mongoexception.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from flask import abort
+from functools import wraps
+from mongoengine.errors import ValidationError
+import logging
+
+def validation(func): # not entirely satisfied by the name, hoping someone will find a better one
```
Comment (Benvii) [inline server/scratchy_server/filters/mongoexception.py:6]: Rename `func` with `ressource_method`.
Add pydoc to explain it should be use to wrappe a MethodView method.

File: server/scratchy_server/filters/mongoexception.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    @wraps(func)
+    def new_func(*args, **kwargs):
```
Comment (Benvii) [inline server/scratchy_server/filters/mongoexception.py:9]: Rename to `mongo_safe_ressource_method`

**Review by Benvii — CHANGES_REQUESTED** [review]
Add this docator to all ressources.

**Pull request closed without merging**